File: Codeforces/1676C-Most_Similar_Words.py
import sys

def solve(arr, n, m):
    min = sys.maxsize

    if n == 2:
        s1 = arr[0]
        s2 = arr[1]
        total = 0

        for i in range(m):
            total += abs(ord(s1[i]) - ord(s2[i]))
        return total

    for i in range(n):
        for j in range(i + 1, n):
            total = 0
            for k in range(m):
                total += abs(ord(arr[i][k]) - ord(arr[j][k]))
            if total < min: min = total
    return min

for _ in range(int(input())):
    n, m = map(int, input().split())
    arr = []
    for _ in range(n):
        arr.append(input())

    print(solve(arr, n, m))

File: Codeforces/1676C-Most_Similar_Words.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <limits.h>

char** init(int n, int m) {
  char** A = (char**) malloc(n * sizeof(char*));
  for (int i = 0; i < n; i++) 
    A[i] = (char*) malloc((m + 1) * sizeof(char));

  for (int i = 0; i < n; i++) {
    fgets(A[i], m + 1, stdin);
    A[i][strcspn(A[i], "\n")] = 0;
  }
  return A;
}

void del(char **A, int n) {
  for (int i = 0; i < n; i++) 
    free(A[i]);
  free(A);
}

int solve(char **A, int n, int m) {
  if (n == 2) {
    int total = 0;
    for (int i = 0; i < m; i++) {
      total += abs(A[0][i] - A[1][i]);
    }

    return total;
  }

  int min = INT_MAX;

  for (int i = 0; i < n; i++) {
    for (int j = i + 1; j < n; j++) {
      int total = 0;
      for (int k = 0; k < m; k++) {
        total += abs(A[i][k] - A[j][k]);
      }
      if (total < min) min = total;
    }
  }

  return min;
}

int main() {
  int t;
  scanf("%d", &t);

  while (t--) {
    int n, m;
    scanf("%d %d", &n, &m);
    char **A = init(n, m);

    printf("%d\n", solve(A, n, m));
    del(A, n);
  }
}
